src/routes/private/kvasir.ts

```diff
@@ -0,0 +1,63 @@
+import type { FastifyInstance } from 'fastify';
+import { Post, Source, SourceMember } from '../../entity';
+import createOrGetConnection from '../../db';
+import { queryReadReplica } from '../../common/queryReadReplica';
+import { Brackets } from 'typeorm';
+import { SourceMemberRoles } from '../../roles';
+import { z } from 'zod';
+
+const postsSchema = z.object({
+  postIds: z.array(z.string()).min(1),
+});
+
+export const kvasir = async (fastify: FastifyInstance): Promise<void> => {
+  fastify.addHook('preHandler', async (req, res) => {
+    if (!req.userId) {
+      return res.code(401).send({ error: 'Unauthorized' });
+    }
```

Comment:
Do you also need:
```
   if (!req.service) {
      return res.status(404).send();
    }
```

just so it can be accessed only internally

Reply:
It already should only be accessed internally, as it's on the /p/ prefix, which is a separate deployment.

On Kvasir side, I just forward the users own auth token, but as cookie instead

Reply:
yeah but just for safety sake, if we have it in other private endpoints let's add here. 

seen the user token part, thats cool 👌

Reply:
Added